Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md

```diff
@@ -0,0 +1,110 @@
+# Upgrading python interpreter version from 3.7.10 to 3.8.12
+
+Currently, in Oppia we are utilizing python interpreter version 3.7.10. We are now in a phase of switching to a more updated interpreter version 3.8.12. For effective and effortless switching to the newer version, this guide will provide a brief and detailed instruction on how to achieve just that. Please follow the below mentioned instructions carefully.
+
+## Prerequisite
+- This page is dedicated to people who want to upgrade their python interpreter in the existing Oppia directory instead of maintaing a new local copy of oppia.
+- Please make sure your local as well as remote fork is up-to-date with the develop. 
+
+## Step 1. Installing python 3.8.12 through pyenv
+Our first step is to install the new interpreter version `3.8.12` through pyenv by issuing the following command through a terminal
+```
+pyenv install 3.8.12
+```
+After you execute the above command you can expect some output as depicted below :
+```
+❯ pyenv install 3.8.12
+Installing Python-3.8.12...
+patching file Misc/NEWS.d/next/Build/2021-10-11-16-27-38.bpo-45405.iSfdW5.rst
+patching file configure
+patching file configure.ac
+Installed Python-3.8.12 to /home/user/.pyenv/versions/3.8.12
+```
+
+## Step 2 (a). Setting up virtualenv for Direnv Users
+*`Note` : Follow this step if you are using direnv for automatic switching of interpreter versions on entering oppia directory.*
+
+`Important` Inorder to know whether you are using direnv or not please carry out the following instructions.
+
+* Go to the directory where you have installed Oppia and check if you have `.envrc` file in that directory. If the `.envrc` file is present, that means you are using direnv for virtualenv activation.
+
+* In the case you are using `direnv` __please follow the below steps to setup virtualenv activation with direnv. Else if you are not a direnv user you can follow instructions from [Step 2 (b)](#step-2-(b).-setting-up-virtualenv-for-pyenv-users).__
+
+#### Steps :
+
+1. Edit the `.envrc` file in your oppia folder to this
+```
+use python 3.8.12
+```
+
+2. Run this command in the same directory to complete the setup
+```
+$ direnv reload
+```
+
+## Step 2 (b). Setting up virtualenv for pyenv users
+*`Note` : Only follow this if you are not using direnv for automatic virtualenv activation*
+
+`Important` Inorder to know whether you are using pyenv or not for virtualenv activation please carry out the following instructions.
```

Comment:
See above re putting this determination into its own separate step and not conflating it with the actual setup step.

Reply:
done